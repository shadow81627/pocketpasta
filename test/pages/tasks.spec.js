import Component from '@/pages/tasks.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import BootstrapVuePlugin from 'bootstrap-vue';

const localVue = createLocalVue();

localVue.use(Vuetify);
localVue.use(BootstrapVuePlugin);

const factory = () =>
  shallowMount(Component, {
    localVue,
  });

describe('Tasks page', () => {
  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('head', () => {
    expect(Component.head()).toBeTruthy();
  });
});
