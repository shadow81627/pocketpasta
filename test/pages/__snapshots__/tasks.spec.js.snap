// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks page renders properly 1`] = `
"<b-container-stub tag=\\"div\\">
  <b-row-stub tag=\\"div\\">
    <b-col-stub tag=\\"div\\">
      <h1>Tasks</h1>
      <v-data-table headers=\\"[object Object],[object Object],[object Object],[object Object]\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]\\" search=\\"\\" sort-by=\\"due\\" item-key=\\"name\\" class=\\"elevation-1\\"></v-data-table>
    </b-col-stub>
  </b-row-stub>
</b-container-stub>"
`;
